input-explainer.md

```diff
@@ -176,16 +176,30 @@ Most applications will want to visually represent the input sources somehow. The
+
+ The `XRInputSource`'s `renderId` is used to determine what should be rendered in the cases when an application specific model isn't being used. 
+
+  - A `renderId` of `''` (empty string) indicates the input source should not be rendered. Cursors and pointing rays should still be rendered as described above if needed by the application. Input sources with a `targetRayMode` of `'gaze'` or `'screen'` MUST have a `renderId` of `''`. `'tracked-pointer'` input sources should typically only have a `renderId` of `''` when using displays that allow the user to see their environment and the physical input source.
+  - The `renderId` may be `'unknown'` if the type of input source cannot be reliably identified or the UA determines that the input source type must be masked for any reason.
+  - Inline sessions MUST only expose `renderId`s of `''` or `'unknown'`.
+  - Otherwise the `renderId` should be a lower-case string that describes the physical input source. It should not include and indication of the handedness of the input source, as that should be implied by the `handedness` attribute.
+    - If the input source is a tracked hand without any held controller, the `renderId` MUST be `hand`
+    - For most controllers this should ideally be of the format `<vendor>-<product-id>`. For example: `oculus-touch`. UAs should make an effort to align on the strings that are returned for any given device.
+
+The WebXR Device API currently does not offer any way to retrieve renderable resources that represent the input devices from the API itself, and as such the `renderId` must be used as a key to load an appropriate resources from the application's server or a CDN. The example below presumes that the `getControllerMesh` call would do the required lookup and caching.
+
 ```js
 // These methods presumes the use of a fictionalized rendering library.
 
 // Render a visualization of the input source - eg. a controller mesh.
 function renderInputSource(session, inputSource, inputPose) {
-  // FIXME: Using a fictional isDisplayOpaque() method to state that controller meshes should not be rendered
-  // on transparent displays (AR).
-  if (isDisplayOpaque(session) && inputPose.gripTransform) {
+  // Controller meshes should not be rendered on transparent displays (AR), so
+  // only render a controller mesh if the XREnvironmentBlendMode is 'opaque' and
+  // we have a valid gripTransform to transform it with.
+  if (inputPose.gripTransform && session.environmentBlendMode == 'opaque') {
```

Comment:
Do we need the `session.environmentBlendMode == 'opaque'` check here?  Per some of @NellWaliczek's comments above, we may be expecting UAs to empty out the `renderId` when in `immersive-ar` mode, at which point this would just be a `inputSource.renderId` check.

On the other hand, if we expect apps to do this kind of check anyway, that could help us be consistent with `renderId`, allowing AR apps to still retrieve the controller model, which they may want anyway for physics (i.e. displacing scene objects) or as a prop for tutorials, even if they're not rendering the controller at the user's hand.